modules/xz/5.4.5.bcr.1/patches/patch.diff

```diff
@@ -0,0 +1,5231 @@
++    "unsupported-filter_flags-1.xz",
++    "unsupported-filter_flags-2.xz",
++    "unsupported-filter_flags-3.xz",
++]
++
++# Files have to be copied from the tests/files directory to the files directory, as the tests expect them here, and bazel maintains the structure from the module root. This is the most universal solution I've found that doesn't involve modifying the tests.
++genrule(
++    name = "copied_files",
++    srcs = ["tests/files/" + f for f in FILE_LIST],
++    outs = ["files/" + f for f in FILE_LIST],
++    cmd_bash = "mkdir -p $(@D)/files && cp $(SRCS) $(@D)/files",
++    # cmd_bat = "mkdir $(@D)\\files >NUL 2>NUL && for %i in ($(SRCS)) do copy /Y %i $(@D)\\files",
++    # cmd_bat = "mkdir $(@D)\\files >NUL 2>NUL && copy /Y tests/files/good-1-check-none.xz $(@D)\\files && copy /Y tests/files/unsupported-check.xz $(@D)\\files",
++    cmd_bat = "echo 'Batch command executed'",
++)
++
++# Filegroup with all copied files
++filegroup(
++    name = "all_test_files",
++    srcs = [":copied_files"],
++    visibility = ["//visibility:public"],
++)
++
++# Test header files
++cc_library(
++    name = "tests_hdrs",
++    testonly = True,
++    hdrs = [
++        "tests/tests.h",
++        "tests/tuktest.h",
++    ],
++)
++
++# Suite of tests, much tidier using the custom macro from the tools.bzl
++test_suite(
++    name = "all_tests",
++    tests = multi_test(
++        size = "small",
++        data = [
++            ":all_test_files",
++        ],
++        names = [
++            "bcj_test",
++            "create_compress_files",
++            "test_bcj_exact_size",
++            "test_block_header",
++            "test_check",
++            "test_filter_flags",
++            "test_filter_str",
++            "test_hardware",
++            "test_index",
++            "test_index_hash",
++            "test_lzip_decoder",
++            "test_memlimit",
++            "test_stream_flags",
++            "test_vli",
++        ],
++        deps = [
++            ":lzma",
++            ":tests_hdrs",
++        ],
++    ) + [
++        # "ossfuzz", # Currently disabled, as the fuzzer doesn't seem to be reliably available for linking on all systems. I don't have experience in this area :)
++    ],
++)
++
++# Fuzz test, not currently implemented
++cc_test(
++    name = "ossfuzz",
++    srcs = ["tests/ossfuzz/fuzz.c"],
++    copts = [
++        "-fsanitize=fuzzer",
++    ],
++    data = [":all_test_files"],
++    linkopts = [
++        "-fsanitize=fuzzer",
++    ],
++    deps = [
++        ":tests_hdrs",
++        "//:lzma",
++    ],
++)
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 0000000..5d1512f
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,8 @@
++module(
++    name = "xz",
++    version = "5.4.5.bcr.1",
++    compatibility_level = 1,
++)
++
++bazel_dep(name = "platforms", version = "0.0.8")
++bazel_dep(name = "bazel_skylib", version = "1.5.0")
+diff --git a/config.lzma-android.h b/config.lzma-android.h
```

Comment:
Is `config.lzma-android.h` generated? Can we write a build rule to generate this file instead of checking it in?

Reply:
Yes, the config files are generated. I considered the possibility of a custom rule to generate, it was a while ago I looked but I think automake is required to generate which might pose a problem. Will hopefully have a chance to look further into it in a couple of weeks, pretty busy rn 😝

Reply:
SG, thanks!

Reply:
FYI, there is a new [cmake_configure_file](https://github.com/bazelbuild/bazel-central-registry/pull/1454) module just checked in BCR, not sure if it's gonna be helpful.

/cc @wep21